extensions/fail.md

```diff
@@ -0,0 +1,50 @@
+---
+title: "`FAIL` Message"
+layout: spec
+copyrights:
+  -
+    name: "Daniel Oaks"
+    period: "2019"
+    email: "daniel@danieloaks.net"
+---
+
+This document specifies the `FAIL` message, intended to provide a clean, consistent interface for sending general errors to clients. We want implementors to be able to provide appropriate error information for newly-developed commands and functions without having to reserve new numerics.
+
+The `FAIL` message may indicate a complete failure to process a given command or function, or simply something about how the processing goes that clients should be aware of.
+
+
+## Message Format
+
+This message has the following format:
+
+    FAIL <command> [<argument>...] <error-code> <description>
+
+Here's what each parameter means, along with a longer breakdown:
+
+- `<command>`: The particular command that spawned the `FAIL` message, or `*` for messages initiated outside client commands.
+- `<argument>`: One or more parameters that give extra information as to where/why the error occurred.
+- `<error-code>`: A machine-readable code for the error encountered. Should stay the same across server and services vendors.
+- `<description>`: A plain-text description of the error encountered, so that users know what occurred. May differ between server and services vendors
+
+The `<command>` parameter is the primary way that clients discover which of their commands has resulted in an `FAIL` response. In addition, this message may be given a label corresponding with the related command, if one is provided by the client. This parameter is case-insensitive. If the `FAIL` message was not triggered by a particular command sent by the client, `*` indicates that this is the case.
```

Comment:
Add link to the label spec, otherwise it may sound ambiguous what you mean here and/or add an example

Reply:
Definitely will add a link here, thanks for the yell